File: src/main/java/gargoyle/netsquares/model/player/DumbPlayer.java
package gargoyle.netsquares.model.player;

import gargoyle.netsquares.model.Board;
import gargoyle.netsquares.model.Direction;
import gargoyle.netsquares.model.Directions;
import gargoyle.netsquares.model.Move;
import gargoyle.netsquares.model.Player;
import org.jetbrains.annotations.NotNull;

public class DumbPlayer extends Player {
    public DumbPlayer(String name) {
        super(name);
    }

    @Override
    protected Move move(@NotNull Board board, @NotNull Directions directions) {
        boolean canOpen = board.canOpen();
        int openScore = board.getAtCaret();
        int max = canOpen ? openScore : Integer.MIN_VALUE;
        Move move = canOpen ? Move.OPEN : null;
        for (Direction direction : directions.getDirections()) {
            for (int i = 1; board.canMoveBy(direction, i); i++) {
                if (board.canOpenBy(direction, i)) {
                    int score = board.getBy(direction, i);
                    if ((max == score && move == null) || max < score) {
                        max = score;
                        move = Move.forDirection(direction);
                    }
                }
            }
        }
        return canOpen && max == openScore ? Move.OPEN : move;
    }
}

File: src/main/kotlin/gargoyle/netsquares/model/player/DumbPlayer.kt
package gargoyle.netsquares.model.player

import gargoyle.netsquares.model.Board
import gargoyle.netsquares.model.Directions
import gargoyle.netsquares.model.Move
import gargoyle.netsquares.model.Player

class DumbPlayer(name: String) : Player(name) {
    override fun move(board: Board, directions: Directions): Move? {
        val canOpen = board.canOpen()
        val openScore = board.atCaret
        var max = if (canOpen) openScore else Int.MIN_VALUE
        var move = if (canOpen) Move.OPEN else null
        for (direction in directions.getDirections()) {
            var i = 1
            while (board.canMoveBy(direction, i)) {
                if (board.canOpenBy(direction, i)) {
                    val score = board.getBy(direction, i)
                    if (max == score && move == null || max < score) {
                        max = score
                        move = Move.forDirection(direction)
                    }
                }
                i++
            }
        }
        return if (canOpen && max == openScore) Move.OPEN else move
    }
}
